src/main/java/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/machines/electric/AutomatedCraftingChamber.java

```diff
@@ -192,40 +198,61 @@ public boolean isSynchronized() {
 			}
 		});
 	}
-	
-	protected void tick(Block b) {
-		if (BlockStorage.getLocationInfo(b.getLocation(), "enabled").equals("false")) return;
-		if (ChargableBlock.getCharge(b) < getEnergyConsumption()) return;
-		
-		BlockMenu menu = BlockStorage.getInventory(b);
-		
+
+	protected void tick(Block block, boolean craftLast) {
+		if (!craftLast) {
+			if (BlockStorage.getLocationInfo(block.getLocation(), "enabled").equals("false")) return;
+		}
+
+		if (ChargableBlock.getCharge(block) < getEnergyConsumption()) return;
+
+		String input = getSerializedInput(block, craftLast);
+		testInputAgainstRecipes(block, input);
+	}
+
+	private String getSerializedInput(Block block, boolean craftLast) {
+		BlockMenu menu = BlockStorage.getInventory(block);
 		StringBuilder builder = new StringBuilder();
 		int i = 0;
+		boolean lastIteration = false;
 		for (int j = 0; j < 9; j++) {
 			if (i > 0) {
 				builder.append(" </slot> ");
 			}
-			
+
 			ItemStack item = menu.getItemInSlot(getInputSlots()[j]);
-			if (item != null && item.getAmount() == 1) return;
+			if (craftLast) {
+				// we're only executing the last possible shaped recipe
+				// we don't want to allow this to be pressed instead of the default timer-based
+				//   execution to prevent abuse and auto clickers
+				if (item != null && item.getAmount() == 1) lastIteration = true;
```

Comment:
I don't really get why you changed this and introduced this variable.
It does pretty much the same as it did before.

Reply:
Previously it would return from the whole method if it encountered an ItemStack with amount 1. However what I was after is getting at least an ItemStack with amount 1 so I could count this as a craftable, last iteration.

Reply:
I also refactored them to look better @ 667ec59

Reply:
I see, alright then!